src/Plugins/DBFTPlugin/Consensus/ConsensusContext.cs

```diff
@@ -199,16 +199,19 @@ public void Reset(byte viewNumber)
                 Snapshot?.Dispose();
                 Snapshot = neoSystem.GetSnapshotCache();
                 uint height = NativeContract.Ledger.CurrentIndex(Snapshot);
+                var isFaun = neoSystem.Settings.IsHardforkEnabled(Hardfork.HF_Faun, height + 1);
                 Block = new Block
                 {
                     Header = new Header
                     {
+                        Version = isFaun ? (uint)BlockVersion.V1 : (uint)BlockVersion.V0,
                         PrevHash = NativeContract.Ledger.CurrentHash(Snapshot),
                         Index = height + 1,
                         NextConsensus = Contract.GetBFTAddress(
                             NeoToken.ShouldRefreshCommittee(height + 1, neoSystem.Settings.CommitteeMembersCount) ?
                             NativeContract.NEO.ComputeNextBlockValidators(Snapshot, neoSystem.Settings) :
-                            NativeContract.NEO.GetNextBlockValidators(Snapshot, neoSystem.Settings.ValidatorsCount))
+                            NativeContract.NEO.GetNextBlockValidators(Snapshot, neoSystem.Settings.ValidatorsCount)),
+                        PrevStateRoot = isFaun ? StateService.StatePlugin.GetStateRootHash(height) : null
```

Comment:
will this null here cause any issue? is the codebase robust enough to tolerate a null prevstateroot?

Reply:
Yes, it's checked.